kstring.c

```diff
@@ -204,8 +204,17 @@ char *kstrtok(const char *str, const char *sep_in, ks_tokaux_t *aux)
 		for (p = start; *p; ++p)
 			if (aux->tab[*p>>6]>>(*p&0x3f)&1) break;
 	} else {
-		for (p = start; *p; ++p)
-			if (*p == aux->sep) break;
+		// Using strchr is fast for next token, but slower for
+		// last token due to extra pass from strlen.  Overall
+		// on a VCF parse this func was 146% faster with // strchr.
+		// Equiv to:
+		// for (p = start; *p; ++p) if (*p == aux->sep) break;
+
+		// NB: We could use strchrnul() here from glibc if detected,
+		// which is ~40% faster again, but it's not so portable.
+		// i.e.   p = (uint8_t *)strchrnul((char *)start, aux->sep);
+		uint8_t *p2 = (uint8_t *)strchr((char *)start, aux->sep);
+		p = p2 ? p2 : start + strlen((char *)start);
```

Comment:
I realise there's a faster alternative here too.

Drop the `if (*p == 0)` code below (so move to the earlier if-else cases) and use p2 == NULL instead.  Eg:

```
uint8_t *p2 = (uint8_t *)strchr((char *)start, aux->sep);
if (p2) {
    aux->p = (const char *)p2;
} else {
    aux->p = (const char *)start + strlen((char *)start);
    aux->finished = 1;
}
```

This avoids a second conditional on `*p`.  Isolated testing of kstrtok though shows it's only around 3-4% quicker so I can't be bothered to change it now, and that corresponds to a tiny amount in most VCF parsing.

Note the old kstrtok was about 18% slower, so the use of strchr has been significant.  We probably ought to apply this to klib, but I see there hasn't been movement on the last PR I made so I suspect support is dead.